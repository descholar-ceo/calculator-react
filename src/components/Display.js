import PropTypes from 'prop-types';

const Display = ({ result }) => <h1>{result}</h1>;
Display.propTypes = {
  result: PropTypes.string.isRequired,
};

export default Display;
