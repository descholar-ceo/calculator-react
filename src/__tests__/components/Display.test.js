import React from 'react';
import renderer from 'react-test-renderer';
import Display from '../../components/Display';

describe('Test Display component with snapshot', () => {
  it('Renders Display component correctly', done => {
    const displayComponent = renderer.create(<Display result="445" />).toJSON();

    expect(displayComponent).toMatchSnapshot();
    done();
  });
});
