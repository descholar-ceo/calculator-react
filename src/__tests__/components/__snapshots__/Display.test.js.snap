// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Display component with snapshot Renders Display component correctly 1`] = `
<h1
  className="result-display"
>
  445
</h1>
`;
